src/Collector.cs

```diff
@@ -153,5 +143,26 @@ private static Socket Connect()
       _log.Info("Connected to collector.");
       return socket;
     }
+
+    private static bool IsSocketDisconnected (Socket socket)
+    {
+      // Is there any data available?
+      byte[] buffer = null;
+      while (socket.Poll(1, SelectMode.SelectRead))
+        {
+          // If no data is available then socket disconnected
+          if (socket.Available == 0)
+            return true;
+
+          // Clear socket data; we don't care what InstrumentApp sends
+          buffer = buffer ?? new byte[Math.Min(1024, socket.Available)];
+          do
+            {
+              socket.Receive(buffer);
+            }
+          while (socket.Available != 0);
```

Comment:
I mean, I think on the second send the server will close the connection.

On Thursday, April 28, 2016, Nathan D Acuff <notifications@github.com>
wrote:

> In src/Collector.cs
> <https://github.com/Instrumental/Instrumental.NET/pull/4#discussion_r61526845>
> :
>
> > +    {
> > +      // Is there any data available?
> > +      byte[] buffer = null;
> > +      while (socket.Poll(1, SelectMode.SelectRead))
> > +        {
> > +          // If no data is available then socket disconnected
> > +          if (socket.Available == 0)
> > +            return true;
> > +
> > +          // Clear socket data; we don't care what InstrumentApp sends
> > +          buffer = buffer ?? new byte[Math.Min(1024, socket.Available)];
> > +          do
> > +            {
> > +              socket.Receive(buffer);
> > +            }
> > +          while (socket.Available != 0);
>
> You are right about that, but that wasn't the situation. The agent was
> just sending hello+authenticate over and over, and the oks were piling up.
> It should not be a problem now.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/Instrumental/Instrumental.NET/pull/4/files/71e433c010fcd20a21ab41cce81d43be4af75643#r61526845>
>